numpy/core/numeric.py

```diff
@@ -374,13 +378,85 @@ def extend_all(module):
         if a not in adict:
             __all__.append(a)
 
+
+def count_nonzero(a, axis=None):
+    """
+    Counts the number of non-zero values in the array ``a``.
+
+    Parameters
+    ----------
+    a : array_like
+        The array for which to count non-zeros.
+
+    axis : int, optional
+        Axis along which to count non-zeros. Default is None,
+        meaning that non-zeros will be counted along a flattened
+        version of ``a``.
+
+        .. versionadded:: 1.12.0
+
+    Returns
+    -------
+    count : int or array of int
+        Number of non-zero values in the array along a given axis.
+        Otherwise, the total number of non-zero values in the array
+        is returned.
+
+    See Also
+    --------
+    nonzero : Return the coordinates of all the non-zero values.
+
+    Examples
+    --------
+    >>> np.count_nonzero(np.eye(4))
+    4
+    >>> np.count_nonzero([[0,1,7,0,0],[3,0,0,2,19]])
+    5
+    >>> np.count_nonzero([[0,1,7,0,0],[3,0,0,2,19]], axis=0)
+    array([1, 1, 1, 1, 1])
+    >>> np.count_nonzero([[0,1,7,0,0],[3,0,0,2,19]], axis=1)
+    array([2, 3])
+
+    """
+    a = asanyarray(a)
+
+    if a.dtype.name == 'bool':
+        if axis is None:
+            # multiarray's count_nonzero
+            # has an optimized version for
+            # booleans on a flat array
+            return multiarray.count_nonzero(a)
+
+        return a.sum(axis=axis)
+
+    nonzeros = (a != 0)
```

Comment:
Oh, okay, I see what you mean.  Empty strings, arrays, etc. should be counted as zero's.  So it really isn't counting zeros then?  It's more counting how many False's we get when iterating over "elements" of the array?